lsp-mode.el

```diff
@@ -4773,7 +4788,10 @@ Applies on type formatting."
                           :text-document (lsp--text-document-identifier)
                           :options (lsp-make-formatting-options
                                     :tab-size (symbol-value (lsp--get-indent-width major-mode))
-                                    :insert-spaces (if indent-tabs-mode :json-false t))
+                                    :insert-spaces (if indent-tabs-mode :json-false t)
+                                    :trim-trailing-whitespace? lsp-trim-trailing-whitespace
```

Comment:
wrap each of these with `lsp-json-bool`  (you can use the same for use-spaces).

Note that in emacs nil will be serialized to null, not false.